### PR TITLE
Seeded ICE when using PDP

### PR DESCRIPTION
**Pull request by renesass**
When passing a seed into PDP, it is not further passed to ICE. This PR solves this.

### CODE REVIEW

**Comment by dwoiwode** [comment]
I understand your issue, but I do not like storing the seed in the class.

Does setting `seed=True` in the ice-property solve this issue? Theoretically this should take the seeded config-space from the PDP object

**Comment by dwoiwode** [comment]
Any insights whether this helped with your issue?

**Comment by renesass** [comment]
Unfortunately, I think the seed has to be stored inside the class because of its design.
It is fact that PDP always have different results although I passed a seed (see https://github.com/automl/DeepCAVE/blob/development/deepcave/plugins/hyperparameter/pdp.py#L197).

**Comment by dwoiwode** [comment]
@renesass Please check, whether #4 is suitable for your issue. Otherwise please provide a test case or a minimal working example that simulates your issue so it can be fixed.

**Comment by renesass** [comment]
Nop.
Tests with `git+https://github.com/dwoiwode/py-pdp-partitioner@3-deterministic-pdp#egg=pyPDPPartitioner`.

I literally do only the following:
`
surrogate_model = RandomForestSurrogate(run.configspace)
surrogate_model.fit(X, Y)

pdp = PDP.from_random_points(
    surrogate_model,
    selected_hyperparameter=selected_hyperparameters,
    seed=0,
    num_grid_points_per_axis=GRID_POINTS_PER_AXIS,
    num_samples=num_samples,
)
`

Proof:
Screenshot 2022-04-28 at 11 56 08
Screenshot 2022-04-28 at 11 57 31

**Comment by renesass** [comment]
Any news here? :)

**Comment by dwoiwode** [comment]
> Nop. Tested with `git+https://github.com/dwoiwode/py-pdp-partitioner@3-deterministic-pdp#egg=pyPDPPartitioner`.
> 
> I literally do only the following:
> 
> ```
> surrogate_model = RandomForestSurrogate(run.configspace)
> surrogate_model.fit(X, Y)
> 
> pdp = PDP.from_random_points(
>     surrogate_model,
>     selected_hyperparameter=selected_hyperparameters,
>     seed=0,
>     num_grid_points_per_axis=GRID_POINTS_PER_AXIS,
>     num_samples=num_samples,
> )
> ```
> 
> Proof: Screenshot 2022-04-28 at 11 56 08 Screenshot 2022-04-28 at 11 57 31

You did not set a seed for your surrogate model in this example. In PDP the (seeded) config-space from the surrogate is taken. Is there a use case where you want to have an unseeded surrogate and a seeded PDP?

**Comment by renesass** [comment]
No, it does not solve the problem because the seed is still not propagated.

Two runs:
RandomForestSurrogate (DEBUG): RandomForestSurrogate: Initial Seed = 0
RandomForestSurrogate (DEBUG): RandomForestSurrogate: Final Seed = 248400791
PDP (DEBUG): PDP: Initial Seed = 0
PDP (DEBUG): PDP: Final Seed = 1543538891
ICE (DEBUG): ICE: Initial Seed = None
ICE (DEBUG): ICE: Final Seed = 1831717508

RandomForestSurrogate (DEBUG): RandomForestSurrogate: Initial Seed = 0
RandomForestSurrogate (DEBUG): RandomForestSurrogate: Final Seed = 248400791
PDP (DEBUG): PDP: Initial Seed = 0
PDP (DEBUG): PDP: Final Seed = 1543538891
ICE (DEBUG): ICE: Initial Seed = None
ICE (DEBUG): ICE: Final Seed = 1831742211

**Comment by dwoiwode** [comment]
> ICE (DEBUG): ICE: Initial Seed = None

This is weird. The initial seed should be set to `True` in this example (See [this line](https://github.com/dwoiwode/py-pdp-partitioner/blob/79544bbe8fd5d3e99c193405bb20755a0e4411b8/pyPDP/algorithms/pdp.py#L38)). 

I just added a test with your provided code and had to change a few lines (e.g. the surrogate is not from this library) but could not reproduce this behaviour. Can you tell me where the difference is to your code? You can find the new test-case [here](https://github.com/dwoiwode/py-pdp-partitioner/blob/8f28c0c80d5bc5449a0258a4b898957525537b68/tests/algorithms/test_pdp.py#L118).

A minimal working example from your side would really help me to address this issue.

**Comment by dwoiwode** [comment]
As no more info is provided, I will close this PR. Feel free to reopen if the issue still persists
